src/bun.js/api/BunObject.zig

```diff
@@ -5297,6 +5302,100 @@ comptime {
     }
 }
 
+pub const JSZstd = struct {
+    export fn zstd_reader_deallocator(_: ?*anyopaque, ctx: ?*anyopaque) void {
+        var reader: *zstd.ZstdReaderArrayList = bun.cast(*zstd.ZstdReaderArrayList, ctx.?);
+        reader.list.deinit(reader.list_allocator);
+        reader.deinit();
+    }
+
+    export fn zstd_compressor_deallocator(_: ?*anyopaque, ctx: ?*anyopaque) void {
+        var compressor: *zstd.ZstdCompressorArrayList = bun.cast(*zstd.ZstdCompressorArrayList, ctx.?);
+        compressor.list.deinit(compressor.list_allocator);
+        compressor.deinit();
+    }
+
+    pub fn zstdDecompressSync(
+        globalThis: *JSGlobalObject,
+        buffer: JSC.Node.StringOrBuffer,
+        options_val_: ?JSValue,
+    ) JSValue {
+        var opts = zstd.Options{};
+        if (options_val_) |options_val| {
+            if (options_val.isObject()) {
+                if (options_val.get(globalThis, "windowLog")) |window| {
+                    opts.window_log_max = window.coerce(i32, globalThis);
+                }
+            }
+        }
+
+        const compressed = buffer.slice();
+        const allocator = JSC.VirtualMachine.get().allocator;
+        var list = std.ArrayListUnmanaged(u8).initCapacity(allocator, if (compressed.len > 512) compressed.len else 32) catch unreachable;
```

Comment:
nit: why the two different limits? It seems to me that compressed.len between 32 and 512 is a potentially common case and we should use compressed.len as capacity then rather than 32.